crates/bls12381/src/curves/g1.rs

```diff
@@ -127,6 +127,34 @@ impl<F: PrimeField + PrimeFieldBits> G1Point<F> {
         Ok(Self { x: xr, y: yr })
     }
 
+    //Implements constraint: (y_a + y_c) * (x_b - x_a) - (y_b - y_a)*(x_a - x_c) = 0 mod p
+    //used to show (x_a, y_a), (x_b, y_b), (x_c, -y_c) are co-linear
+    pub fn collinearcheck<CS>(&self, cs: &mut CS, b: &Self, c: &Self) -> Result<(), SynthesisError>
```

Comment:
To be closer to the naming of the other functions, can we change this to something like:
```suggestion
    pub fn assert_collinear<CS>(cs: &mut CS, a: &Self, b: &Self, c: &Self) -> Result<(), SynthesisError>
```
(this also lets you get rid of the `let (a, b, c) = (self, b, c);` line below)